### PR TITLE
Add KeOps implementations for Matern & TensorProduct

### PR DESCRIPTION
**Pull request by timweiland**
EDIT: Should only be merged once probabilistic-numerics/probnum#793 is merged

### CODE REVIEW

File: src/linpde_gp/functions/_polynomial.py [hunk 1]
```diff
@@ -5,6 +5,10 @@
 import functools
 import itertools
 
+try:
+    from pykeops.numpy import LazyTensor, Pm
+except ImportError:
+    pass
```
Comment (marvinpfoertner) [inline src/linpde_gp/functions/_polynomial.py:11]:
```suggestion
from pykeops.numpy import LazyTensor, Pm
```

File: src/linpde_gp/randprocs/covfuncs/linfuncops/diffops/_matern.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 import functools
+from typing import Optional
 
-from jax import numpy as jnp
-import numpy as np
-from probnum.randprocs import covfuncs
+try:
+    from pykeops.numpy import LazyTensor, Pm, Vi, Vj
+except ImportError:
+    pass
```
Comment (marvinpfoertner) [inline src/linpde_gp/randprocs/covfuncs/linfuncops/diffops/_matern.py:9]:
```suggestion
from pykeops.numpy import LazyTensor, Pm, Vi, Vj
```

File: src/linpde_gp/randprocs/covfuncs/linfuncops/diffops/_tensor_product.py [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
+from typing import Optional
 
-from jax import numpy as jnp
 import numpy as np
+from jax import numpy as jnp
+
+try:
+    from pykeops.numpy import LazyTensor, Pm, Vi, Vj
+except ImportError:
+    pass
```
Comment (marvinpfoertner) [inline src/linpde_gp/randprocs/covfuncs/linfuncops/diffops/_tensor_product.py:12]:
```suggestion
from pykeops.numpy import LazyTensor, Pm, Vi, Vj
```

**Review by marvinpfoertner — APPROVED** [review]
Looks good! 🙏🏻 

Can you add `pykeops` as a requirement in `pyproject.toml`?

**Comment by timweiland** [comment]
Done! Thanks :pray:

**Pull request merged**